zeroex/orderwatch/order_watcher.go

```diff
@@ -211,6 +226,27 @@ func (w *Watcher) cleanupLoop(ctx context.Context) error {
 	}
 }
 
+func (w *Watcher) removedCheckerLoop(ctx context.Context) error {
+	start := time.Now()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+
+		start = time.Now()
+		if err := w.permanentlyDeleteStaleRemovedOrders(ctx); err != nil {
+			return err
+		}
```

Comment:
We check for stale `removed` orders on the leading edge of the loop so that stale orders are removed on Mesh startup.